src/main/java/com/WebController/EventController.java

```diff
@@ -51,16 +74,72 @@ public String receivePatient (@RequestBody List<Event> events){
                 eventRepo.save(event);
             }
         }
-        return "success";
+        return new ResponseEntity<>("saved", HttpStatus.OK);
     }
 
+    /**
+     * DoGet handler for all events.
+     * User will get a list of all existing events with number of attended patients
+     */
     @GetMapping(value = "/allEvents", produces = "application/json")
     @ResponseBody
-    public List<Event> getAllEvents (){
-        List<Event> list = new ArrayList<>();
-        Iterable<Event> events = this.eventRepo.findAll();
+    public Iterable<Event> getAllEvents (@RequestParam("page") int pageNum, ServletResponse response){
+        Iterable<Event> events = this.eventRepo.findAll( PageRequest.of(pageNum-1, 5));
+        Iterable<Event> nextPage = this.eventRepo.findAll( PageRequest.of(pageNum, 5));
+        HttpServletResponse httpServletResponse = (HttpServletResponse) response;
+        if (IterableUtil.sizeOf(nextPage)>0){ //have next page
+            httpServletResponse.addHeader("has-next-page", "true");
+        } else {
+            httpServletResponse.addHeader("has-next-page", "false");
+        }
+        List<Event> list=new ArrayList<>();
         events.forEach(list::add);
         return list;
     }
 
+    @PostMapping(value = "/userregister", consumes = "application/json")
+    public ResponseEntity<String> userRegistration(@RequestBody User user, HttpServletRequest request){
+        if (this.userRepository.findUserByEmailAddressIgnoreCase(user.getEmailAddress())!=null){
+            return new ResponseEntity<>("user already exists", HttpStatus.FORBIDDEN);
+        }
+        userRepository.save(user);
+        ConfirmationToken token = new ConfirmationToken(user);
+        String rootAddress = request.getRequestURL().toString().replace(request.getRequestURI(), request.getContextPath());
+        this.messageSender.sendRegisterEmail(user.getEmailAddress(), token.getConfirmationToken(), rootAddress);
+        this.confirmationTokenRepository.save(token);
+        return new ResponseEntity<>("Confirmation email sent to: "+user.getEmailAddress(), HttpStatus.OK);
```

Comment:
We shouldn't rely on the client processing the string that we're returning. Think in terms of a REST API, with an application that may not be human and read the string.